scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/EcdhPskServerKeyExchange.java

```diff
@@ -0,0 +1,231 @@
+
+import java.net.InetSocketAddress;
+import java.security.GeneralSecurityException;
+import java.security.KeyFactory;
+import java.security.interfaces.ECPublicKey;
+import java.security.spec.ECParameterSpec;
+import java.security.spec.ECPoint;
+import java.security.spec.ECPublicKeySpec;
+import java.util.Arrays;
+
+import org.eclipse.californium.elements.util.DatagramReader;
+import org.eclipse.californium.elements.util.DatagramWriter;
+import org.eclipse.californium.elements.util.StringUtil;
+import org.eclipse.californium.scandium.dtls.AlertMessage.AlertDescription;
+import org.eclipse.californium.scandium.dtls.AlertMessage.AlertLevel;
+import org.eclipse.californium.scandium.dtls.cipher.ECDHECryptography;
+import org.eclipse.californium.scandium.dtls.cipher.ECDHECryptography.SupportedGroup;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * class EcdhPskServerKeyExchange.<br>
+ * This class EcdhPskServerKeyExchange represents {@link ServerKeyExchange} in DHE_PSK mode. The <a
+ * href="https://tools.ietf.org/html/rfc5489#section-2">RFC 5489</a> also define that if server must send the {@link ServerKeyExchange}
+ * message even if the PSK identity hint is not provided. 
+ * @author Vikram
+ *
+ */
+public final class EcdhPskServerKeyExchange extends ServerKeyExchange{
+	
+	private static final Logger LOGGER = LoggerFactory.getLogger(EcdhPskServerKeyExchange.class.getCanonicalName());
+	
+	private static final String MSG_UNKNOWN_CURVE_TYPE = "Unknown curve type [{}]";
+	private static final int CURVE_TYPE_BITS = 8;
+	private static final int NAMED_CURVE_BITS = 16;
+	private static final int PUBLIC_LENGTH_BITS = 8;
+	
+	/**
+	 * The algorithm name to generate elliptic curve keypairs. See also <a href=
+	 * "http://docs.oracle.com/javase/7/docs/technotes/guides/security/StandardNames.html#KeyPairGenerator"
+	 * >KeyPairGenerator Algorithms</a>.
+	 */
+	private static final String KEYPAIR_GENERATOR_INSTANCE = "EC";
+
+	/** The ECCurveType */
+	
+	/** parameters are conveyed verbosely; underlying finite field is a prime 
+	 * field
+	*/ 
+	private static final int EXPLICIT_PRIME = 1;
+	
+	/** parameters are conveyed verbosely; underlying finite field is a 
+	 * characteristic-2 field
+	*/ 
+	private static final int EXPLICIT_CHAR2 = 2;
+	
+	/** a named curve is used */
+	private static final int NAMED_CURVE = 3;
+
+	/** ephemeral public key */
+	private ECPublicKey publicKey = null;
+
+	private ECPoint point = null;
+	private byte[] pointEncoded = null;
+
+	private final int curveId;
+	
+	// TODO right now only named curve is supported
+	private int curveType = NAMED_CURVE;
+	
+	/**
+	 * Constructor {@link EcdhPskServerKeyExchange} called on server side. The {@link EcdhPskServerKeyExchange} 
+	 * message structure is composed of an optional parameter opaque psk identity and ServerDHParams 
+	 * that includes the type of ec curve used and the server public key (point encoded).
+	 * @param ecdhe - {@link ECDHECryptography}
```

Comment:
no need to add a link, JavaDoc automatically generates a link to the parameter's type. Please provide a description of the parameter's role/purpose ...